### PR TITLE
Add Trino to DJQS, including an optional trino docker compose profile

### PR DESCRIPTION
**Pull request by samredai**
### Summary

This adds the official trino client to DJQS to allow sqlalchemy to process `trino://....` URIs. It also includes a couple of bug fixes related to passing in connection args. Lastly, it adds an optional trino container to the docker compose setup that only runs when you include `--profile trino` to the docker compose up command. The official trino docker container comes with the tpch catalog and the sf1 schema that has some nice tables we can use as part of a demo for modeling over trino tables.

### Test Plan

- [x] PR has an associated issue: #1064 
- [x] `make check` passes
- [x] `make test` shows 100% unit test coverage

### CODE REVIEW

File: datajunction-query/.env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CONFIGURATION_FILE="/code/config.djqs.yml"
+DEFAULT_REFLECTION_ENGINE=trino
```
Comment (shangyian) [inline datajunction-query/.env:2]: Can this be simplified to `DEFAULT_ENGINE`?

**Review by shangyian — APPROVED** [review]
Nice! This looks pretty contained too 🥳

File: datajunction-query/.env (hunk 1 above)
Comment (samredai) [reply datajunction-query/.env:2]: This attribute already existed from a previous PR, I just set it here for the demo environment. It's used in a couple of places so I'll follow up with a separate PR to change this everywhere.

Comment (shangyian) [reply datajunction-query/.env:2]: Ah ok, it's fair to just leave it then!

**Pull request merged**